pyomo/contrib/piecewise/transform/outer_representation_gdp.py

```diff
@@ -0,0 +1,112 @@
+    in an "outer" representation--in sets of constraints of the form Ax <= b.
+
+    This transformation can be called in one of two ways:
+        1) The default, where 'descend_into_expressions' is False. This is
+           more computationally efficient, but relies on the
+           PiecewiseLinearFunctions being declared on the same Block in which
+           they are used in Expressions (if you are hoping to maintain the
+           original hierarchical structure of the model). In this mode,
+           targets must be Blocks and/or PiecewiseLinearFunctions.
+        2) With 'descend_into_expressions' True. This is less computationally
+           efficient, but will respect hierarchical structure by finding
+           uses of PiecewiseLinearFunctions in Constraint and Obective
+           expressions and putting their transformed counterparts on the same
+           parent Block as the component owning their parent expression. In
+           this mode, targets must be Blocks, Constraints, and/or Objectives.
+    """
+    CONFIG = PiecewiseLinearToGDP.CONFIG()
+    _transformation_name = 'pw_linear_outer_repn'
+
+    def _transform_pw_linear_expr(self, pw_expr, pw_linear_func,
+                                  transformation_block):
+        transBlock = transformation_block.transformed_functions[
+            len(transformation_block.transformed_functions)]
+
+        # get the PiecewiseLinearFunctionExpression
+        dimension = pw_expr.nargs()
+        transBlock.disjuncts = Disjunct(NonNegativeIntegers)
+        substitute_var = transBlock.substitute_var = Var()
+        pw_linear_func.map_transformation_var(pw_expr,
+                                              substitute_var)
+        substitute_var_lb = float('inf')
+        substitute_var_ub = -float('inf')
+        if dimension > 1:
+            A = np.ones((dimension + 1, dimension + 1))
+            b = np.zeros(dimension + 1)
+            b[-1] = 1
+
+        for simplex, linear_func in zip(pw_linear_func._simplices,
+                                        pw_linear_func._linear_functions):
+            disj = transBlock.disjuncts[len(transBlock.disjuncts)]
+
+            if dimension == 1:
+                # We don't need scipy, and the polytopes are 1-dimensional
+                # simplices, so they are defined by two bounds constraints:
+                disj.simplex_halfspaces = Constraint(
+                    expr=(pw_linear_func._points[simplex[0]][0], 
+                          pw_expr.args[0], 
+                          pw_linear_func._points[simplex[1]][0]))
+            else:
+                disj.simplex_halfspaces = Constraint(range(dimension + 1))
+                # we will use scipy to get the convex hull of the extreme pts of
+                # the simplex
+                extreme_pts = []
+                for idx in simplex:
+                    extreme_pts.append(pw_linear_func._points[idx])
+                chull = spatial.ConvexHull(extreme_pts)
```

Comment:
I guess that this implements quickhull. It might even be worth becoming its own contributed package. I can see it being very useful in building surrogate models a la Zhang (http://egon.cheme.cmu.edu/Papers/Zhang%20Grossmann%20Pinto%20CRS_Paper.pdf)
Food for thought!

Reply:
Yeah, several of the things in `scipy.spatial`, including `scipy.spatial.ConvexHull`, are wrappers around Qhull, which implements quickhull...